src/main/scala/epic/logo/ArgmaxInferencer.scala

```diff
@@ -0,0 +1,7 @@
+package epic.logo
+
+trait ArgmaxInferencer[T, Y, W] {
+
+  def argmax(weights : Weights[W], instance : T) : (Y, FeatureVector[W], Double)
```

Comment:
more or less :)

Hopefully it won't get out that our obscure little outfit is using structured max margin methods!